cpp/src/arrow/json/parser.cc

```diff
@@ -102,7 +104,7 @@ Status Kind::ForType(const DataType& type, Kind::type* kind) {
     Status Visit(const BinaryType&) { return SetKind(Kind::kString); }
     Status Visit(const LargeBinaryType&) { return SetKind(Kind::kString); }
     Status Visit(const TimestampType&) { return SetKind(Kind::kString); }
-    Status Visit(const FixedSizeBinaryType&) { return SetKind(Kind::kString); }
+    Status Visit(const DecimalType&) { return SetKind(Kind::kNumberOrString); }
```

Comment:
Ok, but FixedSizeBinaryType should still be handled separately? Is it tested at all?

Reply:
Oh, it's not supported actually. Fair enough.